erc20-token-bridge/contracts/Bridge.sol

```diff
@@ -79,72 +112,72 @@ contract Bridge is IBridge {
             _signature.s
         );
 
-        if (!isWrapped) {
-            Token(_depositData.token).transferFrom(
-                msg.sender,
-                address(this),
-                _depositData.amount
-            );
-        } else {
-            Token(wrappedToken).burnFrom(msg.sender, _depositData.amount);
-        }
+        Token(_depositData.token).burnFrom(msg.sender, _depositData.amount);
 
-        emit Deposit(
+        emit Burned(
             msg.sender,
             _depositData.to,
             _depositData.token,
             _depositData.targetBridge,
-            _depositData.amount,
-            block.number
+            _depositData.amount
         );
     }
 
-    function withdraw(
+    function unlock(
         WithdrawData calldata _withdrawData
     )
         external
+        payable
+        override
+        nonReentrant
+        validateFee
+        validateToken(_withdrawData.token)
+        validateAmount(_withdrawData.amount)
     {
-        address wrappedToken = tokenToWrappedToken[_withdrawData.token];
-        bool isWrapped = wrappedToken != address(0);
-
-        if (!isWrapped) {
-            Token(_withdrawData.token).transfer(
-                msg.sender,
-                _withdrawData.amount
-            );
-        } else {
-            Token(wrappedToken).mint(msg.sender, _withdrawData.amount);
-        }
+        Token(_withdrawData.token).transfer(msg.sender, _withdrawData.amount);
 
-        emit Withdraw(msg.sender, _withdrawData.token, _withdrawData.amount);
+        emit Unlocked(msg.sender, _withdrawData.token, _withdrawData.amount);
     }
 
-    function increaseBalance(
-        address to,
-        address token,
-        uint256 amount
-    ) external isBridge {
-        balance[to][token] += amount;
+    function mint(
+        WithdrawData calldata _withdrawData
+    )
+        external
+        payable
+        override
+        nonReentrant
+        validateFee
+        validateToken(_withdrawData.token)
+        validateAmount(_withdrawData.amount)
+    {
+        Token(_withdrawData.token).mint(msg.sender, _withdrawData.amount);
+
+        emit Minted(msg.sender, _withdrawData.token, _withdrawData.amount);
     }
 
-    function addBridge(address _bridge) external isAdmin {
+    function addBridge(address _bridge) external onlyOwner {
         bridges[_bridge] = true;
     }
 
-    function addToken(address _token) external isAdmin {
+    function addToken(address _token) external onlyOwner {
         supportedTokens[_token] = true;
     }
 
     function createWrappedToken(
         address token,
         string calldata name,
         string calldata symbol
-    ) external isAdmin {
+    ) external onlyOwner {
         Token wrappedToken = new Token(name, symbol, address(this));
         tokenToWrappedToken[token] = address(wrappedToken);
+        supportedTokens[address(wrappedToken)] = true;
+    }
```

Comment:
This should not be in a separate function but inside the `mint` function. When you call `mint`, you should check if there is already a deployed wrapped token for the native token and re-use it, otherwise deploy a new one and save it in the mapping. You had the same logic before.